base/strings/string.jl

```diff
@@ -18,17 +18,18 @@ const ByteArray = Union{Vector{UInt8},Vector{Int8}}
 """
     String(v::AbstractVector{UInt8})
 
-Create a new `String` from a vector `v` of bytes containing
-UTF-8 encoded characters.
-
-When possible, the data in `v` will be used directly rather than making
-a copy. This is the case for `Vector{UInt8}` arrays returned by [`take!`](@ref)
-on a writable [`IOBuffer`](@ref) or by [`read(io, nb)`](@ref). Else, a copy is made.
-In all cases, one should assume that the function takes "ownership" of the array
-and that its contents should not be subsequently modified. Implementations may choose
-to resize `v` to a zero length to enforce this (which is the case when `v` is a `Vector`).
-
-If you need to subsequently modify `v`, use `String(copy(v))` instead.
+Create a new `String` from a byte vector `v` containing UTF-8 encoded
+characters. If `v` is a mutable vector, it will be truncated and any future
```

Comment:
truncated to zero length

Reply:
Also, this statement seems false.  It only truncates `Vector`, not arbitrary mutable subtypes of `AbstractVector`.

Reply:
The behavior for other types should be implemented to match, but yes, at the moment this may not be 100% so so I guess this docstring is aspirational.